File: EXAMEN/EXAMEN 3/civitasJava/src/civitas/SorpresaPorCasaHotel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */ 
package civitas;
 
import java.util.ArrayList;

/**
 *
 * @author mariolopezgonzalez
 */
public class SorpresaPorCasaHotel extends Sorpresa {
    

    /*
    Constructor con parámetros
    Crea una sorpresa para el resto de sorpresas
    Parametros:
      Recibe como parámetros los valores de los atributos: tipo, valor y texto
     */
    SorpresaPorCasaHotel (int valor, String texto){
        super(texto, valor, null);
    }

     /*
    Método aplicarAJugador()
    Parámetros:
      actual: indice del jugador actual
      todos: array que contiene los jugadores
    Efecto:
       Modifica el saldo del jugador con el valor de la sorpresa multiplicado por el numero de casas y hoteles del jugador
     */
    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)) {
            informe(actual, todos);
            int saldo = getValor() * todos.get(actual).cantidadCasasHoteles();
            todos.get(actual).modificarSaldo(saldo);
        }
    }

    @Override
    public String toString() {
        return "SorpresaPorCasaHotel{ texto="+ super.getTexto() +  '}';
    }
    
    
}

File: EXAMEN/EXAMEN 2/civitasRuby/lib/sorpresa_por_casa_hotel.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class Sorpresa_por_casa_hotel < Sorpresa
    def initialize(valor, texto)
       super(texto, valor)
    end

    # Método aplicarAJugador (PorCasaHotel)
    # Parámetros:
    #   actual: indice del jugador actual
    #   todos: array que contiene los jugadores
    # Efecto:
    #   Modifica el saldo del jugador con el valor de la sorpresa multiplicado por el numero de casas y hoteles del jugador
    def aplicar_a_jugador(actual, todos)
      if jugador_correcto(actual, todos)
        informe(actual, todos)
        saldo = valor*(todos[actual].cantidad_casas_hoteles)
        todos[actual].modificar_saldo(saldo)
      end
    end

    def to_s
      return "Sorpresa Por Casa Hotel -> Texto: #{texto}, Valor: #{valor}"
    end

    public_class_method :new

  end
end
